File: frontend/src/types/match.ts
export type MatchStatus = "UPCOMING" | "LIVE" | "FINISHED" | string

export interface Match {
  id: number
  home_team: string
  away_team: string
  home_score: number | null
  away_score: number | null
  half_score: string | null
  match_time: string
  status: MatchStatus
  league: string
  league_id: number | null
  home_team_id: number | null
  away_team_id: number | null
}

export interface MatchListParams {
  league?: string
  date?: string
  status?: MatchStatus
  page?: number
  page_size?: number
}

export interface StandingSnapshot {
  view: string
  team_name: string
  position: number | null
  points: number | null
  played: number | null
  wins: number | null
  draws: number | null
  losses: number | null
  goals_for: number | null
  goals_against: number | null
  goal_diff: number | null
}

export interface MatchStandings {
  home: StandingSnapshot[]
  away: StandingSnapshot[]
}

export interface MatchOdds {
  current: import("@/types/odds").OddsItem[]
  history: Record<string, import("@/types/odds").OddsHistoryPoint[]>
}

export interface MatchDetail {
  match: Match
  standings: MatchStandings
  comparison: import("@/types/team").MatchComparison
  form: { home: import("@/types/analysis").TeamForm; away: import("@/types/analysis").TeamForm }
  h2h: import("@/types/analysis").H2HRecord[]
  injuries: import("@/types/team").MatchInjuries
  odds: MatchOdds
  prediction: import("@/types/analysis").Prediction | null
  briefing: import("@/types/analysis").Briefing | null
  sentiment: null
}

File: backend/app/schemas/match.py
from datetime import datetime
from pydantic import BaseModel

from app.schemas.odds import OddsHistoryPointOut
from app.schemas.analysis import H2HRecordOut, MatchInjuriesOut, PredictionOut, BriefingOut


class MatchOut(BaseModel):
    id: int
    home_team: str
    away_team: str
    home_score: int | None
    away_score: int | None
    half_score: str | None
    match_time: datetime
    status: str
    league: str
    league_id: int | None
    home_team_id: int | None
    away_team_id: int | None

    model_config = {"from_attributes": True}


class BetOptionOut(BaseModel):
    label: str
    odds: float


class BettingMatchOut(BaseModel):
    match_id: int
    home_team: str
    away_team: str
    match_time: datetime
    league: str
    league_id: int
    status: str
    home_team_id: int
    away_team_id: int
    home_score: int | None
    away_score: int | None
    betting_code: str
    spf: dict
    rqspf: dict
    bf: list[BetOptionOut]
    zjq: list[BetOptionOut]
    bqc: list[BetOptionOut]


class BettingResponse(BaseModel):
    date: str
    weekday: str
    matches: list[BettingMatchOut]


class TeamComparisonOut(BaseModel):
    league_label: str
    team_name: str
    fulltime: dict
    halftime: dict


class MatchComparisonOut(BaseModel):
    match_id: int
    home: TeamComparisonOut
    away: TeamComparisonOut


class StandingSnapshotOut(BaseModel):
    view: str
    team_name: str
    position: int | None
    points: int | None
    played: int | None
    wins: int | None
    draws: int | None
    losses: int | None
    goals_for: int | None
    goals_against: int | None
    goal_diff: int | None


class MatchStandingsOut(BaseModel):
    home: list[StandingSnapshotOut]
    away: list[StandingSnapshotOut]


class MatchDetailOut(BaseModel):
    match: MatchOut
    standings: MatchStandingsOut
    comparison: MatchComparisonOut
    form: dict
    h2h: list[H2HRecordOut]
    injuries: MatchInjuriesOut
    odds: dict
    prediction: PredictionOut | None
    briefing: BriefingOut | None
    sentiment: None
